internal/olm/operator/internal/registry_pod_test.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+
+package olm
+
+import (
+	"context"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+// newFakeClient() returns a clientset
+func newFakeClient() kubernetes.Interface {
+	return fake.NewSimpleClientset()
+}
+
+func TestCreateRegistryPod(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Test Registry Pod Suite")
+}
+
+var _ = Describe("RegistryPod", func() {
+	var rp *RegistryPod
+
+	BeforeEach(func() {
+		rp = &RegistryPod{
+			Kubeclient:  newFakeClient(),
+			DBPath:      "/database/index.db",
+			BundleImage: "quay.io/joelanford/example-operator-bundle:0.2.0",
+			Namespace:   "default",
+			GRPCPort:    50051,
+		}
+
+		rp.setDefaults()
+
+	})
+
+	Describe("creating registry pod", func() {
+
+		Context("with valid registry pod values", func() {
+			var (
+				expectedPodName, expectedOutput string
+			)
+
+			BeforeEach(func() {
+				expectedPodName = "example-operator-bundle-index"
+				expectedOutput = "/bin/mkdir -p index.db &&  /bin/opm registry add -d index.db -b quay.io/joelanford/example-operator-bundle:0.2.0 --mode=semver && /bin/opm registry serve -d index.db -p 50051"
```

Comment:
Ditto  above. we should not need to scroll the page to read the code. It goes against good practices.